### PR TITLE
[web] Remove connection close on blur for desktop browsers

### PR DESCRIPTION
**Pull request by nturgut**
Do not close the text editing connection when an input text element is blurred (upon clicking another element on the page) on a desktop browser. Keep the current behavior for mobile browsers.

This change is requested on many issues:
https://github.com/flutter/flutter/issues/49785
https://github.com/flutter/flutter/issues/55042
https://github.com/flutter/flutter/issues/55449

The features which expect the following behavior will be negatively effected:
(1) User enters text to a TextFormField-A
(2) User clicks on another item Element-B (or jumps to next field using tab)
(3) The onFieldSubmitted on TextFormField-A will trigger, since the editing of the field is finished.
The keyboard keys (Enter, Tab) will still trigger onFieldSubmitted.

Tested on: Chrome Desktop, Firefox, Safari Desktop.
Manually tested for a11y on Chrome Desktop.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart [hunk 1]
```diff
@@ -603,33 +603,6 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
...
-    // focus is called.
-    _subscriptions.add(domElement.onBlur.listen((_) {
-      if (domRenderer.windowHasFocus) {
-        // Focus is still on the body. Continue with blur.
-        owner.sendTextConnectionClosedToFrameworkIfAny();
-      } else {
-        // Refocus.
-        domElement.focus();
-      }
-    }));
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/text_editing/text_editing.dart:631]: As discussed in chat, we need to refocus on blur in order for the keyboard to continue to work.

Comment (nturgut) [reply lib/web_ui/lib/src/engine/text_editing/text_editing.dart]: done. thanks for catching this!

**Comment by nturgut** [comment]
Linux Fuchsia bot has ended with infra error, the status is not synced back to github.

**Comment by nturgut** [comment]
Also Mac Engine Drone, Mac Android Debug Engine also failed with infra failures. Results are not updated to github.

All these three tests are not related to current changes path. Current change only effects Flutter Web Engine. Merging the PR.

**Comment by opheliagame** [comment]
coming here from #55449 and wondering what the current approach would be to resolving the issue on mobile browsers. The example mentioned in the issue doesn't seem to work for me on a desktop browser as well, does this happen for other people too?

**Pull request merged**